core/client/src/main/java/org/phenotips/remote/client/internal/RemoteMatchFinder.java

```diff
@@ -82,59 +82,65 @@ public int getPriority()
     }
 
     @Override
-    protected Set<String> getSupportedServerIdList()
+    public Set<String> getSupportedServerIdList()
     {
         return this.getRemotesList();
     }
 
     @Override
-    protected MatchRunStatus specificFindMatches(Patient patient, String remoteId, List<PatientMatch> matchesList)
+    protected Response specificFindMatches(Patient patient, String remoteId, List<PatientMatch> matchesList)
     {
-        // Checking if a patient has a consent for remote matching
-        if (!this.consentManager.hasConsent(patient, REMOTE_MATCHING_CONSENT_ID)) {
-            this.logger.debug("Skipping patient {}. No consent for remote matching", patient.getId());
-            return MatchRunStatus.NOT_RUN;
-        }
+        try {
+            // Checking if a patient has a consent for remote matching
+            if (!this.consentManager.hasConsent(patient, REMOTE_MATCHING_CONSENT_ID)) {
+                this.logger.debug("Skipping patient {}. No consent for remote matching", patient.getId());
+                return Response.status(Response.Status.FORBIDDEN).build();
+            }
 
-        this.logger.debug("Finding remote matches for patient [{}] on server [{}]", patient.getId(), remoteId);
+            this.logger.debug("Finding remote matches for patient [{}] on server [{}]", patient.getId(), remoteId);
 
-        OutgoingMatchRequest request =
-            this.matchingService.sendRequest(patient.getId(), remoteId, ADD_TOP_N_GENES_PARAMETER);
+            OutgoingMatchRequest remoteResponse =
+                this.matchingService.sendRequest(patient.getId(), remoteId, ADD_TOP_N_GENES_PARAMETER, matchesList);
 
-        MatchRunStatus status = checkRequestValidity(request, patient.getId(), remoteId);
-        if (status != MatchRunStatus.OK) {
-            return status;
-        }
-
-        List<RemotePatientSimilarityView> parsedResults = this.matchingService.getSimilarityResults(request);
-        for (RemotePatientSimilarityView result : parsedResults) {
-            PatientMatch match = new CurrentPatientMatch(result, null, remoteId);
-            matchesList.add(match);
-        }
-        return MatchRunStatus.OK;
-    }
+            // If the response is null, the request was never initiated.
+            if (remoteResponse == null) {
+                this.logger.warn("Remote match request to [{}] was never initiated for patient [{}]",
+                    remoteId, patient.getId());
+                return Response.status(Response.Status.NO_CONTENT).build();
+            }
 
-    private MatchRunStatus checkRequestValidity(OutgoingMatchRequest request, String patientId, String remoteId)
```

Comment:
I woudl keep all the processing in a separate method (as it was before), for clarity. It can just return a `Response` instead of `MatchRunStatus`